gulpfile.js

```diff
@@ -153,31 +179,18 @@ gulp.task('test-react-warnings', function() {
     gutil.log(gutil.colors.red.bold(message));
   };
 
-  return new IndexFileStream(require('./lib/index-static.jsx'))
+  return createIndexFileStream()
     .on('end', function() {
       console.warn = oldWarn;
       if (warnings) {
         this.emit('error', new Error('At least one warning was logged.'));
       }
     })
-    .pipe(gulp.dest('./dist'));
+    .on('data', function() {});
```

Comment:
Oh yeah! This has to do with the way that node streams work, and it's done to put the stream in **flowing mode**. Previously, the pipe to `gulp.dest('./dist')` (i.e., the filesystem) was what made node actually read from the stream. If that `.on('data', function() {})` isn't there, then node will just sit on the stream forever, waiting for it to be piped somewhere or be manually read from. You can read more about it at https://nodejs.org/api/stream.html#stream_class_stream_readable.

You can sort of think of that line as the node equivalent of piping to `/dev/null`, I think. At least that's how I read it. Though it would be more helpful for me to have added a comment like this:

```javascript
.on('data', function() { /* Drain the stream. */ });
```

which I should probably just do, since if you were confused by this, I'm sure others will be too.